new-lamassu-admin/src/components/LogsDownloaderPopper.js

```diff
@@ -153,39 +157,36 @@ const LogsDownloaderPopover = ({ name, getTimestamp, logs, title }) => {
     [setRange]
   )
 
-  const downloadLogs = (range, logs) => {
+  const downloadLogs = (range, args, fetchLogs) => {
     if (!range) return
 
-    if (range.from && !range.to) range.to = moment()
+    fetchLogs({
+      variables: {
+        ...args,
+        ...(selectedRadio === RANGE && {
+          ...(range.from && { from: range.from }),
+          ...(range.to && { until: range.to })
+        })
```

Comment:
this is pretty complicated to understand.

Better if you just mutate the object or create something with _.fromPairs

Reply:
Done!

Reply:
@liordino I'm not sure if this new version would work if we pass down null in from and in to. Did you manage to test that?

Reply:
You mean by selecting the ALL option on the radio button group and them passing null to the gql query? Yes, it's working.